### ISSUE TITLE
Make meshing compliant with bempp-cl

### ISSUE THREAD

**Issue by DanielWinklehner** [DanielWinklehner/py_electrodes#4]
Need to look into the meshing for BEMPP. Would be best to make py_electrodes independent of BEMPP, but return a usable mesh for electrode assemblies.
 
File "...\py_electrodes.py", line 320, in get_bempp_mesh
    _vertices = mesh.leaf_view.vertices
AttributeError: 'Grid' object has no attribute 'leaf_view'

**Comment by pweigel** [comment]
Meshio (https://github.com/nschloe/meshio) may be the solution to this. It looks like a lot of people are starting to use it.

**Comment by DanielWinklehner** [comment]
Yes, maybe. We have to already provide it anyway for bempp-cl, so might as well use it.

**Comment by DanielWinklehner** [comment]
Update on this issue: bempp-cl seems to be inconsistent internally as to which version of meshio to use. Their current [environment.yaml](https://github.com/bempp/bempp-cl/blob/master/environment.yaml) lists version 3.2.12. However, the simple Laplace example with Dirichlet boundaries fails at creating the mesh. When I update to meshio 4.0.10 (current latest), it works. However, then, if I want to use gmsh to plot the GridFunction, it tries to export to a mesh version that meshio no longer supports ("gmsh2-ascii"). In meshio 4 the supported formats are "gmsh" and "gmsh22". I manually override file_format in my local bempp-cl installation and then it works. I will submit a bug report with bempp-cl. Until this has been resolved, I will leave the issue open.
